### PR TITLE
Gp 7982/update deployment config ticket processing

### PR DESCRIPTION
**Pull request by ryanfreckleton**
Initial attempt to break Helm chart into two versions, since we'll want to deploy the new Ticket Workers for testing while maintaining the old version for production and patch releases.

### CODE REVIEW

File: charts/gitprime-app/v2.0.0/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+appVersion: "1.0"
```
Comment (toddjtidwell) [inline charts/gitprime-app/v2.0.0/Chart.yaml]: Might as well change this to 2.0 as well

**Review by toddjtidwell — CHANGES_REQUESTED** [review]

**Comment by toddjtidwell** [comment]
I think you're going to want to rename the "latest" directory to 1.0.0.  That way we have a directory per version, and that directory name won't be confusing.  I *THINK* there is even a way to do a "link" from "latest" to the correct version we're treating as the latest dev version.

File: charts/gitprime-app/v2.0.0/Chart.yaml (hunk 1 above)
Comment (ryanfreckleton) [reply charts/gitprime-app/v2.0.0/Chart.yaml]: :sweat_smile:

**Comment by ryanfreckleton** [comment]
Renamed "latest" to "v1.0.0". I couldn't find a way to 'link' a "latest" folder to the appropriate version after about 20 minutes of googling.

File: charts/gitprime-app/v2.0.0/questions.yml [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+
+  - variable: dataPipeline.newRepoWorker.resources.limits.memory
+    type: string
+    required: true
+    group: "Data Pipeline / New Repo Workers"
+    label: New Repo Worker Memory Limit
+    description: The maximum amount of memory a new repository worker can use.
+    default: "8Gi"
+
+  - variable: dataPipeline.newRepoWorker.requeueAOD
```
Comment (dcumberland) [inline charts/gitprime-app/v2.0.0/questions.yml]: RequeueAOD is no longer needed.  I believe this next version of the app "embeds" this decision.  That is due to, in part, a separation of the commit processing "message consumer" and a new AOD "message consumer".

File: charts/gitprime-app/v2.0.0/templates/dataPipeline/_deployment.tpl [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              value: {{ include "helpers.environment.fullName" . | quote }}
+            - name: SYSTEM_POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: SYSTEM_OPERATION_MODE
+              value: {{ quote .templateData.operationMode }}
+            - name: SYSTEM_CONFIG_SERVER_URL
+              value: {{ include "helpers.configServerURL" . }}
+            - name: GP_REQUEUE_AOD
```
Comment (dcumberland) [inline charts/gitprime-app/v2.0.0/templates/dataPipeline/_deployment.tpl]: Ripple effect of removing the AOD requeue config, this env variable can be removed and we should verify that the config jsons no longer look for it.

File: charts/gitprime-app/v2.0.0/templates/dataPipeline/_deployment.tpl [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+              value: {{ quote .templateData.requeueAOD | default true }}
+            - name: GP_LOG_LEVEL
+              value: {{ quote .templateData.logLevel | default "INFO" | lower }}
+          {{- if .templateData.javaOptions }}
+            - name: JAVA_OPTS
+              value: {{ quote .templateData.javaOptions }}
+          {{- end }}
+          {{- if .templateData.maxConcurrentJobs }}
+            - name: GP_MAX_CONCURRENT_JOBS
+              value: {{ quote .templateData.maxConcurrentJobs }}
```
Comment (dcumberland) [inline charts/gitprime-app/v2.0.0/templates/dataPipeline/_deployment.tpl:54]: The config.json will now have separate values for each queue type, each value.  So things like `GP_MAX_CONCURRENT_JOBS` need to be given with the queue type.

In practice, we will only run one queue type per pod in production.  However, we want to be able to run multiple queues in one pod in other environments, which means we will need to separate out each of these per-queue configs.

**Review by dcumberland — CHANGES_REQUESTED** [review]
There are implications of being able to run multiple queues in one JVM that need to get worked out in the helm chart.  Overall, it should simplify what the helm chart has to do, but it does mean possibly reworking structure.

**Review by dcumberland — APPROVED** [review]
LGTM

**Pull request merged**